examples/sample-next-privy-app/src/app/pages/homepage.tsx

```diff
@@ -56,6 +60,35 @@ const HomePage = (): ReactElement => {
         await sendTransaction(clauses);
     }, [sendTransaction, clauses]);
 
+    const { signTypedData } = usePrivy();
+    const { wallets } = useWallets();
+
+    const embeddedWallet = useMemo<ConnectedWallet | undefined>(() => {
+        return wallets.find((wallet) => wallet.walletClientType === 'privy');
+    }, [wallets]);
+    const THOR_CLIENT = ThorClient.at('https://mainnet.vechain.org');
+    const signer = new AbstractAccountSigner(
+        signTypedData,
```

Comment:
can we get this directly inside the AbstractAccountSigner?
We also need to handle signTypedDataCrossApp